Binary files ./2_step_videos-master/app/src/.DS_Store and ./2_step_videos-develop/app/src/.DS_Store differ
Binary files ./2_step_videos-master/app/src/main/.DS_Store and ./2_step_videos-develop/app/src/main/.DS_Store differ
diff -rupN ./2_step_videos-master/app/src/main/AndroidManifest.xml ./2_step_videos-develop/app/src/main/AndroidManifest.xml
--- ./2_step_videos-master/app/src/main/AndroidManifest.xml	2017-08-05 19:27:00.000000000 -0500
+++ ./2_step_videos-develop/app/src/main/AndroidManifest.xml	2017-08-12 14:22:44.000000000 -0500
@@ -15,7 +15,6 @@
 -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.example.android.miwok">
-
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
@@ -41,6 +40,11 @@
         <activity
             android:name=".SavedRoutinesActivity"
             android:label="@string/category_saved_routine" />
+        <activity
+            android:screenOrientation="landscape"
+            android:configChanges="orientation|keyboardHidden"
+            android:name=".DanceDanceVideoPlayer"
+            android:label="DDVP" />
     </application>
 
 </manifest>
diff -rupN ./2_step_videos-master/app/src/main/java/com/example/android/miwok/DanceDanceVideoPlayer.java ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/DanceDanceVideoPlayer.java
--- ./2_step_videos-master/app/src/main/java/com/example/android/miwok/DanceDanceVideoPlayer.java	1969-12-31 18:00:00.000000000 -0600
+++ ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/DanceDanceVideoPlayer.java	2017-08-13 12:27:46.000000000 -0500
@@ -0,0 +1,115 @@
+package com.example.android.miwok;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Handler;
+import android.view.KeyEvent;
+import android.widget.MediaController;
+import android.widget.TextView;
+import android.widget.VideoView;
+
+public class DanceDanceVideoPlayer extends Activity implements MediaPlayer.OnCompletionListener,MediaPlayer.OnPreparedListener {
+
+    public static int UNKNOWN = -1;
+    public static int ERROR = 1;
+    public static int OK = 0;
+
+    public static int PLAY_VIDEO = 0;
+
+    private VideoView mVV;
+    private MediaController mMC;
+
+    private int mVideoState = UNKNOWN;
+
+    @Override
+    public void onCreate(Bundle b) {
+        super.onCreate(b);
+
+        setContentView(R.layout.videoplayer);
+
+        int fileRes=0;
+        Bundle e = getIntent().getExtras();
+        if (e!=null) {
+            fileRes = e.getInt("fileRes");
+        }
+
+        mVV = (VideoView)findViewById(R.id.myvideoview);
+        mVV.setOnCompletionListener(this);
+        mVV.setOnPreparedListener(this);
+
+        if (playFileRes(fileRes)) {
+            mVV.start();
+            mVideoState = OK;
+        } else {
+            mVideoState = ERROR;
+            this.finish();
+        }
+
+        mMC = new MediaController(this) {
+            @Override
+            public void hide() {
+                //Do not hide.
+            }
+            public boolean dispatchKeyEvent(KeyEvent event)
+            {
+                if (event.getKeyCode() == KeyEvent.KEYCODE_BACK && event.getAction() == KeyEvent.ACTION_UP)
+                    ((Activity) getContext()).finish();
+
+                return super.dispatchKeyEvent(event);
+            }
+        };
+        mMC.setAnchorView(mVV);
+        mVV.setMediaController(mMC);
+        String title2 = e.getString("title2");
+        ((TextView) findViewById(R.id.title)).setText(title2);
+
+    }
+
+    protected void onNewIntent(Intent intent) {
+        super.onNewIntent(intent);
+        setIntent(intent);
+        int fileRes = 0;
+        Bundle e = getIntent().getExtras();
+        if (e != null) {
+            fileRes = e.getInt("fileRes");
+        }
+        playFileRes(fileRes);
+    }
+
+    private boolean playFileRes(int fileRes) {
+        if (fileRes==0) {
+            mVV.stopPlayback();
+            return false;
+        } else {
+            mVV.setVideoURI(Uri.parse("android.resource://" + getPackageName() + "/" + fileRes));
+            return true;
+        }
+    }
+
+    @Override
+    public void onCompletion(MediaPlayer mp) {
+        //Nothing to do here.
+    }
+
+    @Override
+    public void onPrepared(MediaPlayer mp) {
+        mp.setLooping(true);
+        mVV.requestFocus();
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                mMC.show(0);
+            }
+        }, 100);
+    }
+
+    @Override
+    public void onPause()  {
+        super.onPause();
+        this.setResult(mVideoState);
+    }
+
+}
diff -rupN ./2_step_videos-master/app/src/main/java/com/example/android/miwok/MovesActivity.java ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/MovesActivity.java
--- ./2_step_videos-master/app/src/main/java/com/example/android/miwok/MovesActivity.java	2017-08-05 19:27:00.000000000 -0500
+++ ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/MovesActivity.java	2017-08-13 12:28:06.000000000 -0500
@@ -16,14 +16,16 @@
 package com.example.android.miwok;
 
 import android.content.Context;
+import android.content.Intent;
 import android.media.AudioManager;
 import android.media.MediaPlayer;
 import android.os.Bundle;
 import android.support.v7.app.AppCompatActivity;
+import android.widget.VideoView;
 import android.view.View;
 import android.widget.AdapterView;
 import android.widget.ListView;
-
+import android.net.Uri;
 import java.util.ArrayList;
 
 public class MovesActivity extends AppCompatActivity {
@@ -34,6 +36,8 @@ public class MovesActivity extends AppCo
     /** Handles audio focus when playing a sound file */
     private AudioManager mAudioManager;
 
+    private Word mSelectedItem = null;
+
     /**
      * This listener gets triggered whenever the audio focus changes
      * (i.e., we gain or lose audio focus because of another app or device).
@@ -72,9 +76,20 @@ public class MovesActivity extends AppCo
         public void onCompletion(MediaPlayer mediaPlayer) {
             // Now that the sound file has finished playing, release the media player resources.
             releaseMediaPlayer();
+            //Launch the video here
+            launchVideo();
         }
     };
 
+    private boolean launchVideo() {
+        Intent videoPlaybackActivity = new Intent(this, DanceDanceVideoPlayer.class);
+        videoPlaybackActivity.putExtra("fileRes", mSelectedItem.getVideoResourceId());
+        videoPlaybackActivity.putExtra("title1", mSelectedItem.getMiwokTranslation());
+        videoPlaybackActivity.putExtra("title2", mSelectedItem.getDefaultTranslation());
+        startActivityForResult(videoPlaybackActivity, DanceDanceVideoPlayer.PLAY_VIDEO);
+        return true;
+    }
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -85,26 +100,26 @@ public class MovesActivity extends AppCo
 
         // Create a list of words
         final ArrayList<Word> words = new ArrayList<Word>();
-        words.add(new Word("1. Basic", "Beginner", R.drawable.number_one, R.raw.number_one));
-        words.add(new Word("2. Promenade", "Beginner", R.drawable.number_two, R.raw.number_two));
-        words.add(new Word("3. Right Turning Basic (Natural)", "Beginner", R.drawable.number_three, R.raw.number_three));
-        words.add(new Word("4. Right Turning Basic (Cross-Body)", "Beginner", R.drawable.number_four, R.raw.number_four));
-        words.add(new Word("5. Promenade Pivot", "Beginner", R.drawable.number_five, R.raw.number_five));
-        words.add(new Word("6. Underarm Turn (Left)", "Beginner", R.drawable.number_five, R.raw.number_five));
-        words.add(new Word("7. Underarm Turn (Right)", "Beginner", R.drawable.number_six, R.raw.number_six));
-        words.add(new Word("8. Wrap (Walkout)", "Beginner", R.drawable.number_seven, R.raw.number_seven));
-        words.add(new Word("9. Wrap (Check Turn)", "Beginner", R.drawable.number_eight, R.raw.number_eight));
-        words.add(new Word("10. Sweetheart (Check Turn Left)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("11. Sweetheart (Check Turn Right)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("12. Grapevine (Closed)", "Beginner", R.drawable.number_nine, R.raw.number_nine));
-        words.add(new Word("13. Grapevine (Backward Hands)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("14. Grapevine (Forward Hands)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("15. Basket Whip", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("16. Shoulder Catch", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("17. Weave (Inside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("18. Weave (Outside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("19. Weave (Outside/Inside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
-        words.add(new Word("20. Side-by-Side Freespins", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten));
+        words.add(new Word("1. Basic", "Beginner", R.drawable.number_one, R.raw.number_one, R.raw.video_number_one));
+        words.add(new Word("2. Promenade", "Beginner", R.drawable.number_two, R.raw.number_two, R.raw.video_number_one));
+        words.add(new Word("3. Right Turning Basic (Natural)", "Beginner", R.drawable.number_three, R.raw.number_three, R.raw.video_number_one));
+        words.add(new Word("4. Right Turning Basic (Cross-Body)", "Beginner", R.drawable.number_four, R.raw.number_four, R.raw.video_number_one));
+        words.add(new Word("5. Promenade Pivot", "Beginner", R.drawable.number_five, R.raw.number_five, R.raw.video_number_one));
+        words.add(new Word("6. Underarm Turn (Left)", "Beginner", R.drawable.number_five, R.raw.number_five, R.raw.video_number_one));
+        words.add(new Word("7. Underarm Turn (Right)", "Beginner", R.drawable.number_six, R.raw.number_six, R.raw.video_number_one));
+        words.add(new Word("8. Wrap (Walkout)", "Beginner", R.drawable.number_seven, R.raw.number_seven, R.raw.video_number_one));
+        words.add(new Word("9. Wrap (Check Turn)", "Beginner", R.drawable.number_eight, R.raw.number_eight, R.raw.video_number_one));
+        words.add(new Word("10. Sweetheart (Check Turn Left)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("11. Sweetheart (Check Turn Right)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("12. Grapevine (Closed)", "Beginner", R.drawable.number_nine, R.raw.number_nine, R.raw.video_number_one));
+        words.add(new Word("13. Grapevine (Backward Hands)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("14. Grapevine (Forward Hands)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("15. Basket Whip", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("16. Shoulder Catch", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("17. Weave (Inside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("18. Weave (Outside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("19. Weave (Outside/Inside)", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
+        words.add(new Word("20. Side-by-Side Freespins", "Intermediate 1", R.drawable.number_ten, R.raw.number_ten, R.raw.video_number_one));
 
 
         // Create an {@link WordAdapter}, whose data source is a list of {@link Word}s. The
@@ -151,6 +166,8 @@ public class MovesActivity extends AppCo
                     // media player once the sound has finished playing.
                     mMediaPlayer.setOnCompletionListener(mCompletionListener);
                 }
+
+                mSelectedItem = word;
             }
         });
     }
diff -rupN ./2_step_videos-master/app/src/main/java/com/example/android/miwok/Word.java ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/Word.java
--- ./2_step_videos-master/app/src/main/java/com/example/android/miwok/Word.java	2017-08-05 19:27:00.000000000 -0500
+++ ./2_step_videos-develop/app/src/main/java/com/example/android/miwok/Word.java	2017-08-12 14:07:23.000000000 -0500
@@ -31,10 +31,13 @@ public class Word {
     private int mAudioResourceId;
 
     /** Image resource ID for the word */
-    private int mImageResourceId = NO_IMAGE_PROVIDED;
+    private int mImageResourceId = NO_RESOUCE_PROVIDED;
+
+    /** the video in the raw file */
+    private int mVideoResourceId = NO_RESOUCE_PROVIDED;
 
     /** Constant value that represents no image was provided for this word */
-    private static final int NO_IMAGE_PROVIDED = -1;
+    public static final int NO_RESOUCE_PROVIDED = -1;
 
     /**
      * Create a new Word object.
@@ -60,11 +63,21 @@ public class Word {
      * @param audioResourceId is the resource ID for the audio file associated with this word
      */
     public Word(String defaultTranslation, String miwokTranslation, int imageResourceId,
+                int audioResourceId, int videoResourceId) {
+        mDefaultTranslation = defaultTranslation;
+        mMiwokTranslation = miwokTranslation;
+        mImageResourceId = imageResourceId;
+        mAudioResourceId = audioResourceId;
+        mVideoResourceId = videoResourceId;
+    }
+
+    public Word(String defaultTranslation, String miwokTranslation, int imageResourceId,
                 int audioResourceId) {
         mDefaultTranslation = defaultTranslation;
         mMiwokTranslation = miwokTranslation;
         mImageResourceId = imageResourceId;
         mAudioResourceId = audioResourceId;
+        mVideoResourceId = NO_RESOUCE_PROVIDED;
     }
 
     /**
@@ -92,7 +105,7 @@ public class Word {
      * Returns whether or not there is an image for this word.
      */
     public boolean hasImage() {
-        return mImageResourceId != NO_IMAGE_PROVIDED;
+        return mImageResourceId != NO_RESOUCE_PROVIDED;
     }
 
     /**
@@ -101,4 +114,11 @@ public class Word {
     public int getAudioResourceId() {
         return mAudioResourceId;
     }
+
+    /**
+     * Return the audio resource ID of the word.
+     */
+    public int getVideoResourceId() {
+        return mVideoResourceId;
+    }
 }
\ No newline at end of file
Binary files ./2_step_videos-master/app/src/main/res/.DS_Store and ./2_step_videos-develop/app/src/main/res/.DS_Store differ
diff -rupN ./2_step_videos-master/app/src/main/res/layout/videoplayer.xml ./2_step_videos-develop/app/src/main/res/layout/videoplayer.xml
--- ./2_step_videos-master/app/src/main/res/layout/videoplayer.xml	1969-12-31 18:00:00.000000000 -0600
+++ ./2_step_videos-develop/app/src/main/res/layout/videoplayer.xml	2017-08-13 12:26:22.000000000 -0500
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="fill_parent"
+    android:layout_height="fill_parent">
+    <LinearLayout
+        android:orientation="horizontal"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:layout_gravity="center"
+        android:background="@color/category_numbers"
+        android:gravity="center">
+        <VideoView
+            android:id="@+id/myvideoview"
+            android:layout_height="wrap_content"
+            android:layout_width="match_parent"
+            android:layout_marginBottom="0dp"
+            android:layout_gravity="center"
+            android:gravity="center"/>
+    </LinearLayout>
+    <RelativeLayout
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+        <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_alignParentTop="true"
+            android:gravity="center_horizontal"
+            android:padding="10dp"
+            android:textColor="#ffffff"
+            android:textStyle="bold"
+            android:textSize="18sp"
+            android:id="@+id/title"
+            android:text="Test 123"/>
+    </RelativeLayout>
+</FrameLayout>
\ No newline at end of file
diff -rupN ./2_step_videos-master/app/src/main/res/layout/word_list.xml ./2_step_videos-develop/app/src/main/res/layout/word_list.xml
--- ./2_step_videos-master/app/src/main/res/layout/word_list.xml	2017-08-05 19:27:00.000000000 -0500
+++ ./2_step_videos-develop/app/src/main/res/layout/word_list.xml	2017-08-12 14:08:44.000000000 -0500
@@ -13,9 +13,17 @@
      See the License for the specific language governing permissions and
      limitations under the License.
 -->
-<ListView xmlns:android="http://schemas.android.com/apk/res/android"
-    android:id="@+id/list"
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     android:orientation="vertical"
     android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    android:drawSelectorOnTop="true"/>
\ No newline at end of file
+    android:layout_height="match_parent">
+    <ListView xmlns:android="http://schemas.android.com/apk/res/android"
+        android:id="@+id/list"
+        android:orientation="vertical"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:drawSelectorOnTop="true"/>
+    />
+</LinearLayout>
\ No newline at end of file
Binary files ./2_step_videos-master/app/src/main/res/raw/.DS_Store and ./2_step_videos-develop/app/src/main/res/raw/.DS_Store differ
Binary files ./2_step_videos-master/app/src/main/res/raw/video_number_one.mp4 and ./2_step_videos-develop/app/src/main/res/raw/video_number_one.mp4 differ
